src/sim.rs

```diff
@@ -309,10 +312,33 @@ impl<'a> Sim<'a> {
     /// Executes a simple event loop that calls [step](#method.step) each iteration,
     /// returning early if any host software errors.
     pub fn run(&mut self) -> Result {
+        let steps = self.steps.clone();
+        let (tx, rx) = mpsc::channel();
+        std::thread::spawn(move || {
+            let mut blocked = false;
+            loop {
+                let prev = steps.load(std::sync::atomic::Ordering::Relaxed);
+                // Exit if main thread has.
+                match rx.try_recv() {
```

Comment:
How does this behave when you call run() N times in a row? It looks like you could spawn a ton of threads that don't clean up for 10s.

Reply:
That's a good point, I'll adjust it so it uses `recv_timeout` - that way if sim exits early the background thread will be closed too and it'll clean up straight away